src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -131,17 +133,32 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
                            << status.ToString();
           }
         }
-      }));
-  pending_lease_requests_.insert(scheduling_key);
+      });
+
+  if (status.ok()) {
+    pending_lease_requests_.insert(scheduling_key);
+  } else {
+    if (lease_client != local_lease_client_) {
+      // A remote request failed. Retry the worker lease request locally
+      // if it's still in the queue.
+      // TODO(swang): Fail after some number of retries?
+      RAY_LOG(ERROR) << "Retrying attempt to schedule task at remote node. Error: "
+                     << status.ToString();
+      RequestNewWorkerIfNeeded(scheduling_key);
+    } else {
+      RAY_LOG(FATAL) << "Lost connection with local raylet. Error: " << status.ToString();
+    }
+  }
```

Comment:
The above code looks the same as the error handling in the reply callback of `RequestWorkerLease`, could you please unify them?

Reply:
I'm going to revert this back to the RAY_CHECK_OK since as Eric said, this status is meaningless.